File: src/dataset/cpp/sieve_exercism-tqa236/openAI/ChatGPT4_sieve_v1.cpp
#include "sieve.h"

#include <vector>

namespace sieve {
vector<int> primes(int number) {
  if (number < 2) return {};

  vector<bool> is_prime(number + 1, true);
  is_prime[0] = is_prime[1] = false;
  vector<int> primes;

  for (int i = 2; i * i <= number; ++i) {
    if (is_prime[i]) {
      for (int j = i * i; j <= number; j += i) {
        is_prime[j] = false;
      }
    }
  }

  for (int i = 2; i <= number; ++i) {
    if (is_prime[i]) {
      primes.push_back(i);
    }
  }

  return primes;
}
}  // namespace sieve

File: src/dataset/Java/sieve_exercism-ThomasZumsteg/openAI/ChatGPT4_Sieve_v1.java
import java.util.List;
import java.util.ArrayList;

public class Sieve {
    private boolean[] sieve;

    public Sieve(int limit) {
        sieve = new boolean[limit + 1];
    }

    public List<Integer> getPrimes() {
        List<Integer> primes = new ArrayList<>();
        if (sieve.length > 2) primes.add(2);
        int sqrtLimit = (int) Math.sqrt(sieve.length - 1);
        for (int i = 3; i < sieve.length; i += 2) {
            if (!sieve[i]) {
                primes.add(i);
                if (i <= sqrtLimit) {
                    for (int m = i * i; m < sieve.length; m += i * 2) {
                        sieve[m] = true;
                    }
                }
            }
        }
        return primes;
    }
}